### PR TITLE
Fix bug where the order of the nodes vec was wrong

### PR DESCRIPTION
**Pull request by setzer22**
## Checklist

- CI Checked:
  - [x] `cargo fmt` has been ran
  - [ ] `cargo clippy` reports no issues
  - [ ] `cargo test` succeeds
  - [ ] `cargo rend3-doc` has no warnings
  - [x] `cargo deny check` issues have been fixed or added to deny.toml
- Manually Checked:
  - [x] relevant examples/test cases run
  - [x] changes added to changelog
    - [x] Add credit to yourself for each change: `Added new functionality @githubname`.

## Description
I realized my last PR #340 had a bug. The bug did not affect the example scenes because those are not using the returned `nodes` array, but it came up when I was implementing a test for the animation system.

Basically, I was iterating the node list in topological sort order, but pushing into `final_nodes`, so the values of all parent/children index "pointers" became invalid.

### CODE REVIEW

**Review by cwfitzgerald — APPROVED** [review]
Good catch
